### PR TITLE
[Fix] Small fixes

### PR DESCRIPTION
**Pull request by pomadchin**
- [x] Allowed empty extents (added test coverage)
- [x] Catching cropped tiles to combine
- [x] Remove merge strategy from subproject build files that don't need it

### CODE REVIEW

**Comment by pomadchin** [comment]
@lossyrob Quick question about merge strategy: now we have it only for `spark-etl` and for `spark` sub projects; should I left these lines only for a `spark-etl` or we need `spark` assembly too?

File: vector/src/main/scala/geotrellis/vector/Extent.scala [hunk 1]
```diff
@@ -43,9 +43,9 @@ object ProjectedExtent {
  */
 case class Extent(xmin: Double, ymin: Double, xmax: Double, ymax: Double) {
 
-  // Validation: Do not accept extents min values greater than max values.
-  if (xmin > xmax) { throw ExtentRangeError(s"Invalid Extent: xmin must be less than xmax (xmin=$xmin, xmax=$xmax)")  }
-  if (ymin > ymax) { throw ExtentRangeError(s"Invalid Extent: ymin must be less than ymax (ymin=$ymin, ymax=$ymax)") }
+  // Validation: Do not accept extents min values greater than or equal to max values (allows empty extents).
+  if (xmin >= xmax) { throw ExtentRangeError(s"Invalid Extent: xmin must be less than or equal to xmax (xmin=$xmin, xmax=$xmax)")  }
```
Comment (pomadchin) [inline vector/src/main/scala/geotrellis/vector/Extent.scala]: @lossyrob Want to note, that `xmin >= xmax` *not* allowing empty extents; the previous condition `xmin > xmax` accepted: other words all extents with `xmin <= xmax` were accepted; should I rollback changes?

File: vector-test/src/test/scala/spec/geotrellis/vector/ExtentSpec.scala [hunk 2]
```diff
@@ -95,6 +99,11 @@ class ExtentSpec extends FunSpec with Matchers {
       assert(e.contains(10.0, 10.0) === false)
     }
 
+    it("empty extent should contain empty extent") {
```
Comment (pomadchin) [inline vector-test/src/test/scala/spec/geotrellis/vector/ExtentSpec.scala]: Should empty extent contain an empty extent? How the contain function should behave on extent bounds?

File: vector/src/main/scala/geotrellis/vector/Extent.scala (hunk 1 above)
Comment (lossyrob) [inline vector/src/main/scala/geotrellis/vector/Extent.scala]: Whoops, yes we were already accepting empty extents then, so roll that back.

File: vector-test/src/test/scala/spec/geotrellis/vector/ExtentSpec.scala (hunk 2 above)
Comment (lossyrob) [inline vector-test/src/test/scala/spec/geotrellis/vector/ExtentSpec.scala]: No I don't think an empty extent should contain anything, including itself

**Comment by lossyrob** [comment]
Merge strategy only applies to ETL

File: raster/src/main/scala/geotrellis/raster/ArrayTile.scala [hunk 3]
```diff
@@ -138,7 +138,9 @@ trait ArrayTile extends Tile with Serializable {
       case ct: ConstantTile =>
         ct.combine(this)(f)
       case ct: CompositeTile =>
-        ct.combine(this)((z1, z2)=>f(z2, z1))
+        ct.combine(this)((z1, z2) => f(z2, z1))
+      case ct: CroppedTile =>
```
Comment (lossyrob) [inline raster/src/main/scala/geotrellis/raster/ArrayTile.scala]: Can you add a small test that will cover this code?

File: vector-test/src/test/scala/spec/geotrellis/vector/ExtentSpec.scala (hunk 2 above)
Comment (pomadchin) [inline vector-test/src/test/scala/spec/geotrellis/vector/ExtentSpec.scala]: And what about non empty extents? Should any extent contain itself?

File: raster/src/main/scala/geotrellis/raster/ArrayTile.scala (hunk 3 above)
Comment (pomadchin) [inline raster/src/main/scala/geotrellis/raster/ArrayTile.scala]: :+1: thanks for a quick review

File: vector-test/src/test/scala/spec/geotrellis/vector/ExtentSpec.scala (hunk 2 above)
Comment (lossyrob) [inline vector-test/src/test/scala/spec/geotrellis/vector/ExtentSpec.scala]: yes non empty extents contain themselves

**Pull request merged**